[ios_zip_validation] Validate zip

lailabecker opened the pull request:
https://app.asana.com/0/1199509904376633/1199959636277655/f

Commit:
[ios_zip_validation] Validate zip

https://app.asana.com/0/1199509904376633/1199959636277655/f

tschuls reviewed (changes requested):
I do not agree to solve this logic in the iOS only part, since we have the same problem in Android as well. I will propose another solution.

tschuls committed:
[ios_zip_validation] Merge branch 'develop' into feature/ios_zip_validation

# Conflicts:
#	iOS/Multiplatform Redux Sample/Multiplatform Redux Sample/Resources/en.lproj/Localizable.strings

tschuls committed:
[ios_zip_validation] Only allow valid zips in settings

tschuls committed:
[ios_zip_validation] Only allow valid zips on Android

tschuls committed:
[ios_zip_validation] Use new state property for disabling back navigation

tschuls reviewed (approved):
@Nils-Becker my understanding of the problem is:
- limiting the number of digits to 4
- not allowing back navigation if there is no valid zip

-> I did Add the limitation of 4 digits to android as well, the second point I handled in the State and reused it on both platforms. Is it okay with you?

Nils-Becker commented:
> @Nils-Becker my understanding of the problem is:
> 
> * limiting the number of digits to 4
> * not allowing back navigation if there is no valid zip
> 
> -> I did Add the limitation of 4 digits to android as well, the second point I handled in the State and reused it on both platforms. Is it okay with you?

I think there should be a way to cancel changing the zip code in the settings. Maybe we could have a "save" button in the upper right corner that is only enabled if the entry is valid and navigating back always discards your changes.

Nils-Becker commented:
I think there should be a way to cancel changing the zip code in the settings. Maybe we could have a "save" button in the upper right corner that is only enabled if the entry is valid and navigating back always discards your changes.
